cpp/ql/src/semmle/code/cpp/models/implementations/StdContainer.qll

```diff
@@ -0,0 +1,102 @@
+/**
+ * Provides models for C++ containers `std::array`, `std::vector`, `std::deque`, `std::list` and `std::forward_list`.
+ */
+
+import semmle.code.cpp.models.interfaces.Taint
+
+/**
+ * Additional model for standard container constructors that reference the
+ * value type of the container (that is, the `T` in `std::vector<T>`).  For
+ * example the fill constructor:
+ * ```
+ * std::vector<std::string> v(100, potentially_tainted_string);
+ * ```
+ */
+class StdContainerConstructor extends Constructor, TaintFunction {
+  StdContainerConstructor() {
+    this.getDeclaringType().hasQualifiedName("std", "vector") or
+    this.getDeclaringType().hasQualifiedName("std", "deque") or
+    this.getDeclaringType().hasQualifiedName("std", "list") or
+    this.getDeclaringType().hasQualifiedName("std", "forward_list")
+  }
+
+  /**
+   * Gets the index of a parameter to this function that is a reference to the
+   * value type of the container.
+   */
+  int getAValueTypeParameter() {
+    getParameter(result).getType().getUnspecifiedType().(ReferenceType).getBaseType() =
```

Comment:
```suggestion
    getParameter(result).getUnspecifiedType().(ReferenceType).getBaseType() =
```

We've preferred this shorter form since https://github.com/github/codeql/pull/1343.